start-here.mdx

```diff
@@ -1,8 +1,153 @@
+
+Internal and incoming external messages execute the account's code. The code is interpreted by [TON Virtual Machine](/tvm/overview) (TVM). It is written in _bitcode_, a binary format specific to TVM. In the future, TVM might support multiple binary languages, _codepages_, but at the moment there is only codepage 0 (`CP0`).
+
+Most development is done in the [Tolk](/languages/tolk/overview), a high-level programming language. Its compiler is included in the [Blueprint](/contract-dev/blueprint/overview) development environment and will be available to all projects created from the template.
+
+Originally, [Fift](/languages/fift/overview), a Forth-like assembly language, and [FunC](/languages/func/overview), a C-like intermediate-level language, were used for contract development.
+
+When execution [starts](/tvm/initialization), the message and current account state are provided to the code. By the end of execution, the destination account might change its state or code, or send internal or outgoing external messages.
+
+The execution follows a procedure whose steps are called [phases](/foundations/phases). [Fees](/foundations/fees) are deducted during this process. Fees might be deducted from the account's balance or from the message, depending on the [mode](/foundations/messages/modes) of the message.
+
+Execution cost is first measured in _gas_ units, then converted to Toncoin. This unit is separate so that if code execution becomes computationally cheaper (or more expensive), validators can vote to change the [price of gas](/foundations/config#param-20-and-21:-gas-prices) in Toncoin.
+
+If something goes wrong, a non-zero [exit code](/tvm/exit-codes) might be returned, no changes to state or code are saved, and no further messages are sent.
+
+![Trace overview](/resources/images/trace.png)
+
+The only reason any code can be executed is when some account has received a message. Internal messages can only be sent by another contract executing some code, and that contract must have received a message from somewhere too. In the end, every message can be considered part of some _trace_: a tree of messages between accounts that starts with an incoming external message, continues with internal messages, and possibly ends with some outgoing external messages.
+
+If an external service needs to extract data from the blockchain, it can call a [get method](/tvm/get-method): a handler of incoming external messages in the code of some account. It's called a "method" because it acts like one: the message provides its parameters, and its execution returns a value. Get methods cannot make any changes to the blockchain.
```

Comment:
Let's also state explicitly that contracts cannot call each other's get methods: it is a too common question